src/backend/common/datafeeds/parsers/fms_api/fms_api_awards_parser.py

```diff
@@ -0,0 +1,76 @@
+import json
+from typing import Any, Dict, List, Optional
+
+from google.cloud import ndb
+
+from backend.common.datafeeds.parsers.json.parser_json import ParserJSON
+from backend.common.helpers.award_helper import AwardHelper
+from backend.common.models.award import Award
+from backend.common.models.event import Event
+from backend.common.models.team import Team
+
+
+class FMSAPIAwardsParser(ParserJSON[List[Award]]):
+    def __init__(self, event: Event, valid_team_nums: Optional[List[int]] = None):
+        self.event = event
+        self.valid_team_nums = valid_team_nums
+
+    def parse(self, response: Dict[str, Any]) -> Optional[List[Award]]:
+        awards_by_type = {}
+        for award in response["Awards"]:
```

Comment:
Thinking about this - these methods have 3 returns. Some values, an empty array (valid from the server), or incorrectly formatted data. The parsers right now are responsible for handling the zero data case (empty arrays) in an attempt to not overwrite data, but this feels like something we should be doing in the datafeeds or elsewhere.

We should update these models to remove the `return ... if ... else None` checks and only return `None` if the JSON was incorrect.

Reply:
Nah ya know what - I don't care about this right now. This is a complicated situation that will be more obvious what is the right answer once the datafeed gets brought over. I'm going to leave this as-is for now.